components/header/navbar/searchpage/index.tsx

```diff
@@ -0,0 +1,292 @@
+                      )
+                    ) : (
+                      <p className="text-white mt-4 left-0 text-opacity-70">
+                        Inga resultat hittades för {query}.
+                      </p>
+                    )
+                  }
+                </>
+              )
+          }
+        </div>
+
+        <div className="mt-5 h-full w-full">
+          {!fileSearch && (
+            <>
+              {results.pageData.map((page) => (
+                <ListCard
+                  key={page.attributes.title}
+                  title={page.attributes.title}
+                  setSideMenuOpen={setSideMenuOpen}
+                  path={`/${page.attributes.category?.data.attributes.slug}/${page.attributes.slug}`}
+                  onClick={onClose}
+                  content={page.attributes.content}
+                />
+              ))}
+              {results.sectionData.map(
+                (section) =>
+                  section.attributes?.title &&
+                  section.attributes?.content_page?.data && (
+                    <ListCard
+                      key={section.id}
+                      title={section.attributes.title}
+                      setSideMenuOpen={setSideMenuOpen}
+                      onClick={onClose}
+                      path={`/${section.attributes.content_page?.data?.attributes.category?.data.attributes.slug}/${section.attributes.content_page?.data.attributes.slug}#${titleToAnchor(section.attributes.title ?? '')}`}
+                      content={section.attributes.content}
+                    />
+                  )
+              )}
+              {results.privatePageData.map((page) => (
+                <ListCard
+                  key={page.attributes.title}
+                  title={page.attributes.title}
+                  setSideMenuOpen={setSideMenuOpen}
+                  path={`/medlem/${page.attributes.slug}`}
+                  onClick={onClose}
+                  content={page.attributes.content}
+                />
+              ))}
+              {results.privateSectionData.map(
+                (section) =>
+                  section.attributes.title &&
+                  section.attributes?.private_page?.data && (
+                    <ListCard
+                      key={section.id}
+                      title={section.attributes.title}
+                      setSideMenuOpen={setSideMenuOpen}
+                      onClick={onClose}
+                      path={`/medlem/${section.attributes.private_page?.data.attributes.slug}#${titleToAnchor(section.attributes.title ?? '')}`}
+                      content={section.attributes.content}
+                    />
+                  )
+              )}
+            </>
+          )}
+          {fileSearch && (
+            <>
+              {fileResults.files?.map(
+                (file, idx) =>
+                  file.id &&
+                  file.name && (
+                    <FileCard
```

Comment:
It would maybe be better if `FileCard` would take a `fileResult` and read the needed fields